packages/@aws-cdk/aws-apigatewayv2/lib/http/authorizer.ts

```diff
@@ -15,11 +15,34 @@ export enum HttpAuthorizerType {
 
   /** Lambda Authorizer */
   LAMBDA = 'REQUEST',
+}
+
+/**
+ * Supported Route Authorizer types
+ */
+export enum HttpRouteAuthorizerType {
```

Comment:
Okay they're two values:
- Type of Authorizer (as defined by https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-authorizer.html#cfn-apigatewayv2-authorizer-authorizertype)
- Type of authorization on route (as defined by https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-route.html#cfn-apigatewayv2-route-authorizationtype)

Reply:
Well, that's confusing. 

It would be better to hide this weirdness inside the L2s for Authorizer and Route, and expose just one enum from the CDK. Can we do that?

Reply:
Isn’t it kinda hidden right now?. Remember the user only has to deal with the authorizer and its type. The Route authorization type is being handled internally by the L3 constructs:

- HttpJwtAuthorizer
- HttpLambdaAuthorizer
- HttpUserPoolAuthorizer

Unless i’m not understanding [construct levels](https://aws.amazon.com/blogs/developer/working-with-the-aws-cloud-development-kit-and-aws-construct-library/), we have to expose both of these for L2.

Reply:
Just to get the terminology right. There are no L3 constructs here. There are L2 constructs and integration pattern - I believe these are what you're referring to.

> Isn’t it kinda hidden right now?

It's not because it's exported as part of the public API. 

Let's keep it simple. Since it's only used within the return of the integration classes, i.e., `HttpRouteAuthorizerConfig`, drop this entire type and make the property `authorizationType` as a string.

The other alternative is to use `HttpAuthorizationType`, change it from a string enum to just a standard enum. And compute the value the enum should translate within each implementation. This probably end up more complicated than we need it to be, so let's keep it simple.